lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -31,6 +33,8 @@ ExecutionEngine::ExecutionEngine(BackendKind backendKind) {
 /// Set the code generator kind to \p backendKind.
 void ExecutionEngine::setBackend(BackendKind backendKind) {
   setBackend(createBackend(backendKind));
+  device_ = runtime::DeviceManager::createDeviceManager(backendKind,
```

Comment:
The comment in the header says `Note: at the moment This DOES NOT recreate the DeviceManager`, but you seem to create a new device manager here. Is that comment outdated? Or may be I misunderstand the code here?

If the comment is correct, may be it is a good idea to add an assert checking that the new backendKind is the same as the old one?

Reply:
The comment is on the version of setBackend that takes a Backend* instead of BackendKind. There's currently no way to get the BackendKind of a Backend which means we can't do that check, and we can't just recreate the DeviceManager with the correct BackendKind. I avoided adding it to keep the diff simple, but it's a small enough thing it makes sense to add now I guess.